packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts

```diff
@@ -48,33 +43,18 @@ const FLAGS_REGEX = /^[\da-f]{2}$/;
  *     For more information see {@link https://www.w3.org/TR/trace-context/}
  */
 export function parseTraceParent(traceParent: string): SpanContext | null {
-  const trimmed = traceParent.trim();
-  const traceParentParts = trimmed.split('-');
-
-  // Current version must be structured correctly.
-  // For future versions, we can grab just the parts we do support.
-  if (
-    traceParentParts[0] === VERSION &&
-    traceParentParts.length !== VERSION_PART_COUNT
-  ) {
-    return null;
-  }
+  const match = TRACE_PARENT_REGEX.exec(traceParent);
+  if (!match) return null;
 
-  const [version, traceId, parentId, flags] = traceParentParts;
-  const isValidParent =
-    VERSION_REGEX.test(version) &&
-    TRACE_ID_REGEX.test(traceId) &&
-    PARENT_ID_REGEX.test(parentId) &&
-    FLAGS_REGEX.test(flags);
-
-  if (!isValidParent) {
-    return null;
-  }
+  // According to the specification the implementation should be compatible
+  // with future versions. If there are more parts, we only reject it if it's using version 00
+  // See https://www.w3.org/TR/trace-context/#versioning-of-traceparent
+  if (match[1] === '00' && match[5]) return null;
```

Comment:
Is this covered by some test?

Reply:
Yes, it was just not that explicitly stated in the code https://github.com/marcbachmann/opentelemetry-js/blob/simplify-traceparent-parsing/packages/opentelemetry-core/test/context/HttpTraceContext.test.ts#L148-L159